abci/src/tracing_span.rs

```diff
@@ -38,20 +41,28 @@ macro_rules! block_span {
 /// # Examples
 ///
 /// ```
-/// # use tenderdash_proto::abci::{RequestInfo, request};
+/// # use tenderdash_proto::abci::{RequestInfo, Request, request::Value};
 /// # use tenderdash_abci::tracing_span::span;
 ///
-/// let request = request::Value::Info(RequestInfo::default());
-/// let span = span(request);
+/// let request = Request {
+///    value: Some(Value::Info(Default::default())),
+/// };
+/// let span = span(&request);
 /// ```
-pub fn span<T>(request: T) -> tracing::span::EnteredSpan
+pub fn span(request: &abci::Request) -> tracing::span::EnteredSpan
 where
-    T: Into<Value>,
 {
-    let value = request.into();
+    let value = request.value.as_ref().expect("request value is missing");
+
+    // we use md5 as we need 16-byte request id for uuid, and it doesn't have to be
```

Comment:
This approach has downsides, especially if we want to filter by specific requests in elastic. There are endpoints like info, query, etc. where we don't have much difference so they will have the same ID.
We could have both: unique request ID and md5 so we will benefit from both depending on what we need.

That would be great if we could use the same request ID with tenderdash so we can easily filter related events.